main/src/ca/uwaterloo/flix/language/phase/Lowering.scala

```diff
@@ -540,19 +547,22 @@ object Lowering {
 
     case Expression.NewChannel(exp, tpe, pur, eff, loc) =>
       val e = visitExp(exp)
-      val t = visitType(tpe)
-      Expression.NewChannel(e, t, pur, eff, loc)
+      val unboxedChannelType = Type.mkEnum(Enums.ChannelMpmc, e.tpe :: Nil, loc)
+      val newChannel = Expression.Def(Defs.ChannelNew, Type.mkImpureArrow(Type.Int32, unboxedChannelType, loc), loc)
+      Expression.Apply(newChannel, e :: Nil, unboxedChannelType, pur, eff, loc)
 
     case Expression.GetChannel(exp, tpe, pur, eff, loc) =>
       val e = visitExp(exp)
-      val t = visitType(tpe)
-      Expression.GetChannel(e, t, pur, eff, loc)
+      // val unboxedChannelType = Type.mkEnum(Enums.ChannelMpmc, e.tpe :: Nil, loc)
+      val getChannel = Expression.Def(Defs.ChannelGet, Type.mkImpureArrow(Types.ChannelMpmc, Types.Boxed, loc), loc)
+      Expression.Apply(getChannel, e :: Nil, e.tpe, pur, eff, loc)
```

Comment:
`e.tpe` should be `t` since `e` is the channel expression, and the lowered type should not change, elements are the same type.